pepper-apis/src/main/java/org/broadinstitute/ddp/DataDonationPlatform.java

```diff
@@ -598,4 +649,139 @@ private static void initSqlCommands(Config sqlConfig) {
+            MatcherFilter matcherFilter = new MatcherFilter(routeMatcher, staticFilesConfiguration, exceptionMapper, false, hasMultipleHandler);
+            matcherFilter.init((FilterConfig)null);
+            SessionHandler handler = new JettyCustomRemoteAddrHeaderHandler(matcherFilter, this.preferredIPSourceHeader);
+            handler.getSessionCookieConfig().setHttpOnly(this.httpOnly);
+            return (new EmbeddedJettyServer(this.serverFactory, handler)).withThreadPool(this.threadPool);
+        }
+
+        public AppEngineServerFactory withThreadPool(ThreadPool threadPool) {
+            this.threadPool = threadPool;
+            return this;
+        }
+
+        public AppEngineServerFactory withHttpOnly(boolean httpOnly) {
+            this.httpOnly = httpOnly;
+            return this;
+        }
+    }
+
+    /**
+     * Lifted from {@link spark.embeddedserver.jetty.JettyHandler}, with one
+     * change to use {@link SourceIPHeaderRequestWrapper} to set
+     * the source IP properly in the appengine environment
+     */
+    public static class JettyCustomRemoteAddrHeaderHandler extends SessionHandler {
+        private javax.servlet.Filter filter;
+        private String preferredIPSourceHeader;
+
+        public JettyCustomRemoteAddrHeaderHandler(Filter filter, String preferredIPSourceHeader) {
+            this.filter = filter;
+            this.preferredIPSourceHeader = preferredIPSourceHeader;
+        }
+
+        public void doHandle(String target, org.eclipse.jetty.server.Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+            HttpRequestWrapper wrapper = new SourceIPHeaderRequestWrapper(request, preferredIPSourceHeader);
+            this.filter.doFilter(wrapper, response, null);
+            if (wrapper.notConsumed()) {
+                baseRequest.setHandled(false);
+            } else {
+                baseRequest.setHandled(true);
+            }
+        }
+    }
+
+    /**
+     * Lifted from {@link spark.embeddedserver.jetty.JettyServer}
+     */
+    public static class JettyServer implements JettyServerFactory {
+        JettyServer() {
+        }
+
+        public Server create(int maxThreads, int minThreads, int threadTimeoutMillis) {
+            Server server;
+            if (maxThreads > 0) {
+                int min = minThreads > 0 ? minThreads : 8;
+                int idleTimeout = threadTimeoutMillis > 0 ? threadTimeoutMillis : '\uea60';
+                server = new Server(new QueuedThreadPool(maxThreads, min, idleTimeout));
+            } else {
+                server = new Server();
+            }
+
+            return server;
+        }
+
+        public Server create(ThreadPool threadPool) {
+            return threadPool != null ? new Server(threadPool) : new Server();
+        }
+    }
+
+    public static final class FilterConfig implements javax.servlet.FilterConfig {
```

Comment:
I could not  find an existing implementation  in our various jars, so I rolled this one.